### ISSUE TITLE
How does typescript make generics mandatory 

### ISSUE THREAD

**Issue by laterdayi** [typescript-eslint/typescript-eslint#6901]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

*

### Description

```
export const test = <T>(a:string,{c,d} = {c: T[], D:T} = {})=>{} 
```

How does typescript make generics mandatory

I want to define a function that has a T type so that the developer must pass in the type when using it

### Fail

```typescript
test()
```

### Pass

```typescript
test<string>()  
test<number>()
```

**Comment by JoshuaKGoldberg** [comment]
@laterdayi it looks like you're trying to ask for help on TypeScript in general? This isn't the place for that. You might want to ask in the [TypeScript Discord](https://discord.com/invite/typescript).
